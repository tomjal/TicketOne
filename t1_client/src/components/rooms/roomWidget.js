import React, { Component } from 'react';

import { authManager } from './../../services/authManager';
import ChatWindow from './chat/chatWindow';

export class RoomWidget extends Component {
  constructor() {
    super();
    this.sendMessage = this.sendMessage.bind(this);
    this.state = { isOpened: true };
  }
  sendMessage(message) {
    this.props.sendMessageCallback(message, this.props.id);
  }
  render() {
    const { id, role, messages } = this.props;
    const isClientRoom = authManager.isClient(role);

    return (
      <div className="room-widget">
        {isClientRoom &&
          <div className="widget__header-text">
            Hello {id} - how can we help you?
          </div>
        }
        {!isClientRoom &&
          <div className="widget__header-text">
            id: {id}
          </div>
        }
        <div>
          {this.state.isOpened &&
            <ChatWindow
              role={role}
              sendMessage={this.sendMessage}
              messages={messages} />}
        </div>
      </div>
    );
  }
}

export default RoomWidget;
