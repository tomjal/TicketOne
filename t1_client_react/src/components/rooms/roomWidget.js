import React, { Component } from 'react';

import { authManager } from './../../services/authManager';
import ChatWindow from './chat/chatWindow';

export class RoomWidget extends Component {
  constructor() {
    super();
    this.sendMessage = this.sendMessage.bind(this);
  }
  sendMessage(message) {
    this.props.sendMessageCallback(message, this.props.id);
  }
  markAsResolved = () => {
    this.props.markAsResolvedCallback(this.props.clientId, this.props.id);
  }
  markAsUnresolved = () => {
    this.props.markAsUnresolvedCallback(this.props.clientId, this.props.id);
  }
  render() {
    const { id, role, messages } = this.props;
    const isClientRoom = authManager.isClient(role);

    const chatHeaderText = isClientRoom ? `${this.props.topic}` : `topic: ${this.props.topic}`;
    return (
      <div className="room-widget">
        <div className="widget__header">
          {chatHeaderText}
          {isClientRoom && this.props.isOpened &&
            <p className="widget__header-menu">
              <button className="btn btn-default" onClick={this.markAsResolved}>Mark as Resolved</button>
              <button className="btn btn-default" onClick={this.markAsUnresolved}>Mark as Unresolved</button>
            </p>
          }
          {isClientRoom && !this.props.isOpened &&
            <p className="widget__header-menu">
              Closed and marked as: {this.props.isSolved ? "Solved" : "Unsolved"}</p>
          }
          {!isClientRoom &&
            <div>
              <div>
                room id: {id.toString()}
              </div>
              <div>
                client id: {this.props.creatorId.toString()}
              </div>
            </div>
          }
        </div>
        <div>
          <ChatWindow
            isOpened={this.props.isOpened}
            role={role}
            sendMessage={this.sendMessage}
            messages={messages} />
        </div>
      </div>
    );
  }
}

export default RoomWidget;
