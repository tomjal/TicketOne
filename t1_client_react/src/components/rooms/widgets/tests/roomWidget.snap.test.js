import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { RoomWidget } from './../roomWidget';

test('RoomWidget snapshot', () => {
  const props = {
    id: 0,
    creatorId: 1,
    role: "",
    messages: []
  };

  const wrap = shallow(<RoomWidget {...props} />);

  expect(shallowToJson(wrap)).toMatchSnapshot();
});