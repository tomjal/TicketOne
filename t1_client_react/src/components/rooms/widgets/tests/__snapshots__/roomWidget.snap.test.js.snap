// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomWidget snapshot 1`] = `
<div
  className="room-widget"
>
  <div
    className="widget__header"
  >
    topic: undefined
    <div>
      <div>
        room id: 
        0
      </div>
      <div>
        client id: 
        1
      </div>
    </div>
  </div>
  <div>
    <ChatWindow
      messages={Array []}
      role=""
      sendMessage={[Function]}
    />
  </div>
</div>
`;
